cluster-autoscaler/metrics/liveness.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+)
+
+type HealthCheck struct {
+	lastActivity time.Time
+	mutex        *sync.Mutex
```

Comment:
I don't really mind too much, but you could instead do:
```
type HealthCheck struct {
    sync.Mutex
    lastActivity    time.Time
}
```

And later use `healthCheckInstance.Lock()`.